### ISSUE TITLE
aws_route_table crash on terraform plan

### ISSUE THREAD

**Issue by mkolbert** [hashicorp/terraform#11134]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

0.8.3 as well as 0.8.2 prior to upgrading today.

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_instance
- dnsimple_record

aws_route_table

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```
variable "AWS_REGION" { default = "us-east-1" }

variable "PEERED_ENV" { default = "XXXXXXXX" }

provider "aws" {
  alias = "this"
  region = "${var.AWS_REGION}"
  profile = "${var.PEERED_ENV}"
}

data "aws_route_table" "RTB" {
  provider = "aws.this"
  tags = {
    Name = "Private Route Table"
    Region = "us-east-1a"
  }
}
```
### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

<script src="https://gist.github.com/mkolbert/465e0e5ec4da96b48ff255ba29a1839b.js"></script>

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

<script src="https://gist.github.com/mkolbert/dc6ad90a6bb0647bd286d83597dce00b.js"></script>

### Expected Behavior
What should have happened?

Using same syntax with aws_subnet works fine.

### Actual Behavior
What actually happened?

Crash

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

terraform plan

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

VPC

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234

**Comment by grubernaut** [comment]
Hi @mkolbert, thanks for the issue!

I'm having trouble reproducing the issue you're facing. Is there any other configuration that would allow for a successful reproduction? 

Also would you be able to pass along the output of the following `aws` API call:
```
aws ec2 describe-route-table --route-table-ids rtb-2d84ca4a --region us-west-2
```

Thanks!

**Comment by mkolbert** [comment]
Hi Jake,
The only terraform file I have in my directory is test.tf with the following contents:


# Test

variable "AWS_REGION" { default = "us-east-1" }

variable "PEERED_ENV" { default = "copyright" }

provider "aws" {
  alias = "this"
  region = "${var.AWS_REGION}"
  profile = "${var.PEERED_ENV}"
}

data "aws_route_table" "RTB" {
#data "aws_subnet" "RTB" {
  provider = "aws.this"
  tags = {
#    Name = "AWS Private AZ-1a"
    Name = "Private Route Table"
    Region = "us-east-1a"
  }
}



If I use the same file using the 2 commented out lines in place of the corresponding ones to do the same for subnet I get this:



data.aws_subnet.RTB: Refreshing state...

No changes. Infrastructure is up-to-date. This means that Terraform
could not detect any differences between your configuration and
the real physical resources that exist. As a result, Terraform
doesn't need to do anything.


I have 3 route tables in this account with the following tags:


Route 1
               Name:   Private Route Table
               Region: us-east-1a

Route 2
               Name:   Private Route Table
               Region: us-east-1c

Route 3
               Name:   Default Public Route Table



Here’s the tags returned by the API call for my route tables


aws ec2 describe-route-tables --region us-east-1


            "Tags": [
                {
                    "Value": "us-east-1a",
                    "Key": "Region"
                },
                {
                    "Value": "Private Route Table",
                    "Key": "Name"
                }
            ],



            "Tags": [
                {
                    "Value": "Default Public Route Table",
                    "Key": "Name"
                }
            ],



            "Tags": [
                {
                    "Value": "us-east-1c",
                    "Key": "Region"
                },
                {
                    "Value": "Private Route Table",
                    "Key": "Name"
                }
            ],

I hope that helps.
Matt

**Comment by grubernaut** [comment]
@mkolbert I actually wanted to see the Associations for each Route Table returned from that API call. I believe that we're running into an issue where the Route Table(s) that are being returned to Terraform's `data` resource are the main route table for their respective VPC. This would explain the crash, as the main route table for a VPC does not have an explicit subnet association. 
Thanks!

**Comment by grubernaut** [comment]
@mkolbert I've opened PR#11139 which should fix your issue. Thanks!

**Comment by dstine** [comment]
Confirmed fixed in 0.8.5 by @mkolbert.
